repos/system_upgrade/el7toel8/actors/checkipaserver/actor.py

```diff
@@ -0,0 +1,87 @@
+from leapp.actors import Actor
+from leapp.models import IpaInfo
+from leapp import reporting
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+REPORT_COMMON = [
+    reporting.Tags([reporting.Tags.SERVICES]),
+    reporting.Remediation(
+        hint="Please follow the IdM RHEL 7 to 8 migration guide lines."
+    ),
+    reporting.ExternalLink(
+        url=(
+            "https://access.redhat.com/documentation/en-us"
+            "/red_hat_enterprise_linux/8/html/installing_identity_management"
+            "/migrate-7-to-8_migrating"
+        ),
+        title="Migrating IdM from RHEL 7 to 8",
+    ),
+    reporting.RelatedResource("package", "ipa-server"),
+]
+
+
+class CheckIPAServer(Actor):
+    """Check for ipa-server and inhibit upgrade
+    """
+
+    name = "check_ipa_server"
+    consumes = (IpaInfo,)
+    produces = (reporting.Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for ipainfo in self.consume(IpaInfo):
+            if ipainfo.is_server_configured:
+                self._ipa_inhibit_upgrade(ipainfo)
+            elif ipainfo.has_server_package:
+                self._ipa_warn_pkg_installed(ipainfo)
+
+    def _ipa_inhibit_upgrade(self, ipainfo):
```

Comment:
Please move these code into a library, see: https://github.com/oamg/leapp-repository/tree/master/repos/system_upgrade/el7toel8/actors/rpmscanner  for a example how to use it.

Reply:
I have moved the helper and reporting functions to library file.